### PR TITLE
Bug 1861695: Update defaultGCPDiskImage

### PR DESCRIPTION
**Pull request by enxebre**
This https://github.com/openshift/installer/pull/3808 changed the expectation for the GCP default image. It reuses now published RHCOS images, projects/rhcos-cloud/global/images/*

### CODE REVIEW

File: pkg/apis/machine/v1beta1/machine_webhook.go [hunk 1]
```diff
@@ -113,6 +110,10 @@ const (
 	defaultGCPCredentialsSecret = "gcp-cloud-credentials"
 	defaultGCPDiskSizeGb        = 128
 	defaultGCPDiskType          = "pd-standard"
+	// https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.6/46.82.202007212240-0/x86_64/meta.json
+	// https://github.com/openshift/installer/pull/3808
+	// https://github.com/openshift/installer/blob/d75bf7ad98124b901ae7e22b5595e0392ed6ea3c/data/data/rhcos.json
+	defaultGCPDiskImage = "projects/rhcos-cloud/global/images/rhcos-46-82-202007212240-0-gcp-x86-64"
```
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook.go:116]: Do we expect this to change often?

File: pkg/apis/machine/v1beta1/machine_webhook_test.go [hunk 2]
```diff
@@ -382,7 +382,7 @@ func TestMachineUpdate(t *testing.T) {
 		},
 	}
 
-	vsphereClusterID := "vsphere-cluster"
+	vsphereClusterID := "vsdefaultGCPDiskImagephere-cluster"
```
Comment (JoelSpeed) [inline pkg/apis/machine/v1beta1/machine_webhook_test.go:385]: Ehhh? Bad replace? 😂

Comment (enxebre) [reply pkg/apis/machine/v1beta1/machine_webhook_test.go]: upsss heh

File: pkg/apis/machine/v1beta1/machine_webhook.go (hunk 1 above)
Comment (enxebre) [reply pkg/apis/machine/v1beta1/machine_webhook.go:116]: I don't believe so. We could choose to change it one per release. This is the instance bootimage, the early pivot will take over and get you to the latest os version after bootstrapping a new instance.
If/when MCO makes the image cluster version available we could dynamically default to it.

**Comment by JoelSpeed** [comment]
Spot tests failed but webhook tests passed again, no need to retest GCP as we know the spot tests are broken (https://github.com/openshift/cluster-api-actuator-pkg/pull/185)

/lgtm

**Pull request merged**